docs/internal-states.md

```diff
@@ -0,0 +1,178 @@
+
+- **[getInternalStatesetInternalState(suffix?: string)](https://doc.deno.land/https://deno.land/x/windmill@v1.14.1/index.ts/~/getInternalState)**** allows you to access previously set internal states. The function accepts the optional string parameter **suffix** (suffix of the path of the internal state) can be input as an argument in order to to share internal state between jobs.
+
+## How to use Internal States
+Here is a simple example of accessing a previous state with **getInternalState()**, updating it with **setInternalState()** and then accessing the updated state.
+
+First we will obtain the previously held state on the Windmill Cloud.
+```js
+export async function main(x: number) {
+    const before = await wmill.getInternalState()
+}
+```
+
+The state can then be updated or set using **setInternalState()** by specifying an object with a property and value. 
+```js
+export async function main(x: number) {
+    const before = await wmill.getInternalState()
+    await wmill.setInternalState({test: x})
+}
+```
+
+The updated state can then be accessed and saved to a variable which can be returned. 
+```js
+export async function main(x: number) {
+    const before = await wmill.getInternalState()
+    await wmill.setInternalState({test: x})
+    const after = await wmill.getInternalState()
+    return { before, after }
+}
+```
+## Implementation using Internal States vs Postgres SQL Database
+
+Lets take a look at how internal states can be used to implement a script that allows users to check if a github repo has new stars. First we'll show you how the script is implemented using a postgres database to save the state of the github repo stars value. Then we'll compare it to how the same script is implemented with the power of Windmill internal states.
+
+### Postgres Implementation
+
+The script takes an input argument of a json object containing a Github repo name with it's corresponding star count. The script to generate this input is available on Windmill Hub and can be accessed [here](https://hub.windmill.dev/scripts/github/1206/get-repo-star-count-github#approved). 
+
+```json
+//{"reponame": star_count}
+
+{"awesome":206586}
+```
+
+The below script connects to a Postgres Database and compares the previously held star count value of the repo that is held in the database to the value that was just obtained from input argument. If the new star value is greater than the old star value, the script will return a notification message containing the amount of new stars that the repo received.
+If the repo is not in the database, the script will add it. Additionally, if the repo lost stars the script will update the database. 
+```js
+{
+  export async function main(data: object) {
+    const db = await wmill.getResource("u/User/windmill_postgres_db"); // Postgres Database Resource
+    console.log(db);
+    db.database = db.dbname;
+    db.hostname = db.host;
+    const client = new Client(db);
```

Comment:
I recently released https://doc.deno.land/https://deno.land/x/windmill@v1.15.0/mod.ts/~/databaseUrlFromResource
which should allow to do something like just:
`const client = new Client(await databaseUrlFromResource("u/User/windmill_postgres_db"))`